### ISSUE TITLE
5shot testing results inconsistent when bsz=1 and bsz!=1

### ISSUE THREAD

**Issue by CandiceWu123** [pawn-sxy/DCAMA#10]
Hello, thanks for your excellent work!
During test, when I set nshot=5 and bsz=1, the results did not equal to the results when bsz!=1. I debugged the code and I thought that the reason was due to line 195, line 196 in model/DCAMA.py
`support_feat = torch.stack([support_feats[k][idx] for k in range(nshot)])`
`support_feat = support_feat.view(-1, ch, ha * wa).permute(0, 2, 1).contiguous()`
I found that the shape of support feat was (shot, bsz, ch, h, w) but not (bsz, shot, ch, h, w), when flatten the tensor, support features of the same index in different batches would line together, however when bsz=1 support features of the same batch lined together. I wonder if I understood this part incorrectly or I missed some details.
Thank you!

**Comment by pawn-sxy** [comment]
Hi @CandiceWu123, thank you for the careful debugging, and sorry for the very late response. Your understanding is correct: this is a bug in the multi-shot feature ordering.

The default `torch.stack(..., dim=0)` produces `[shot, batch, C, H, W]`. The subsequent reshape preserves that shot-first ordering, but `MultiHeadedAttention` later regroups the keys using the query batch size. When both `nshot > 1` and `bsz > 1`, this mixes support features from different episodes and misaligns them with the batch-first support masks. With `bsz=1`, the two orderings coincide, so the bug is hidden.

Fixed in [ccab20c](https://github.com/pawn-sxy/DCAMA/commit/ccab20c0ed2002f3592fb88e9dcfd6fb12e9f02a) on `main`. The only change is to stack along `dim=1`:

```python
support_feat = torch.stack([support_feats[k][idx] for k in range(nshot)], dim=1)
```

This produces `[batch, shot, C, H, W]`; the following reshape can remain unchanged, and the existing per-shot positional encoding is preserved.

I verified the fix with the actual PyTorch DCAMA head on both CPU and CUDA, comparing batched inference against processing the same episodes individually. Tests using synthetic features and reduced channel counts reproduced the issue before the fix and passed afterward, including 5-shot batches of sizes 1, 2, and 3. The remaining output differences were below `1e-7`. This was a targeted regression check, not a full pretrained-backbone/dataset evaluation.

Please pull the latest `main` and try again with identical query/support episodes in evaluation mode. Thanks again for identifying the cause!
